### PR TITLE
[web] Disable font face observer on (pre-Chromium) MS Edge

### LINKED ISSUE #5935: Expo vector icons are broken on Edge
### Environment

```
  Expo CLI 3.1.2 environment info:
    System:
      OS: macOS 10.14.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 8.11.3 - /var/folders/qt/10542x9d2wbg0x93mmh1xsw00000gn/T/yarn--1570620719131-0.9749200633049342/node
      Yarn: 1.17.3 - /var/folders/qt/10542x9d2wbg0x93mmh1xsw00000gn/T/yarn--1570620719131-0.9749200633049342/yarn
      npm: 5.6.0 - ~/.asdf/shims/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5791312
      Xcode: 11.0/11A420a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.0
      react: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8
    npmGlobalPackages:
      expo-cli: 3.1.2
```

Issue reproduced on Edge 44 on Windows 10 (both on regular PC and [using official VM](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/))

### Steps to Reproduce

1. Import `MaterialIcons` from `@expo/vector-icons` and try to render one
2. Open the page in Microsoft Edge

### Expected Behavior

Icon is visible

### Actual Behavior

TTF file is loaded but the icon is not showing up.

Additionally, but this is not happening in the reproducible demo, in our case we've got a bunch of "3000ms timeout exceeded" errors in the console.

### Reproducible Demo

https://github.com/d4rky-pl/expo-web-broken-icons

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

fix #5935

# How

- Disable font face observer on MS Edge
- Disable entire module in node

# Test Plan

- should run on edge
- shouldn't throw in SSR envs

### CODE REVIEW

File: packages/expo-font/src/ExpoFontLoader.web.ts [hunk 1]
```diff
@@ -5,14 +5,22 @@ export default {
     return 'ExpoFontLoader';
   },
 
-  loadAsync(fontFamilyName: string, resource: string): Promise<void> {
+  async loadAsync(fontFamilyName: string, resource: string): Promise<void> {
+    if (!('document' in global)) {
```
Comment (ide) [inline packages/expo-font/src/ExpoFontLoader.web.ts]: seems like you should use the canUseDOM check

**Review by ide — CHANGES_REQUESTED** [review]
Edge is moving to Chromium, in which case I'd expect this API to start working. Need to figure out how to distinguish between old and new Edge.

**Comment by EvanBacon** [comment]
Should be fine. You detect new Edge by finding `edg` https://social.msdn.microsoft.com/Forums/en-US/edde0150-9478-4ca6-ba0f-9e084b5a4719/how-to-detect-microsoft-edge-chromium-chrome-ie-browser-using-javascript?forum=iewebdevelopment

**Comment by EvanBacon** [comment]
Saving unrelated ios_build failure: https://app.circleci.com/jobs/github/expo/expo/98499/parallel-runs/0/steps/0-117

**Review by ide — APPROVED** [review]
OK, good to know

**Comment by lucasfronza** [comment]
Had the same issue with Firefox and Safari.

```
if (navigator.userAgent.includes('Firefox')) {
    return;
}
if (navigator.userAgent.includes('Safari')) {
    return;
}
```
seems to fix it.

**Comment by jjg77** [comment]
I'm seeing the same issue in Firefox and Safari.  Is there a fix?

**Pull request merged**